### PR TITLE
fix: add DISPLAY_NAME_LENGTH to be configurable

### PR DESCRIPTION
**Pull request by adong**
## Context

People are frustrated not able to customize job name's length to fit their own needs, and we cannot predict the proper length for each use case, so we are opening up the configuration to users to decide. 

This is just a first step toward building customizing personal preference, which yet to come

## Objective
To enable users to have a customized length of job name in the workflow graph page, by using `localStorage`. 

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by tkyi** [comment]
I still think focusing on displaying job description would let users have shorter job names and resolve this display issue: https://github.com/screwdriver-cd/screwdriver/issues/1493

**Comment by adong** [comment]
> I still think focusing on displaying job description would let users have shorter job names and resolve this display issue: [screwdriver-cd/screwdriver#1493](https://github.com/screwdriver-cd/screwdriver/issues/1493)

Yep, cannot agree with you enough.

File: app/components/workflow-graph-d3/component.js [hunk 1]
```diff
@@ -136,7 +137,14 @@ export default Component.extend({
     });
   },
   draw(data) {
-    const MAX_DISPLAY_NAME = 20;
+    let MAX_DISPLAY_NAME = ENV.APP.DISPLAY_NAME_LENGTH;
+
+    const preferredLength = parseInt(localStorage.getItem('DISPLAY_NAME_LENGTH'), 10);
```
Comment (jithine) [inline app/components/workflow-graph-d3/component.js:142]: can you keep default to current 20

Comment (adong) [reply app/components/workflow-graph-d3/component.js:142]: Default is 20, `10` means base 10 for `parseInt(numberInStringFormat, 10)`,

Comment (jithine) [reply app/components/workflow-graph-d3/component.js:142]: 🤦

**Pull request closed without merging**